eureka2-write-server/src/test/java/com/netflix/eureka2/server/service/bootstrap/PeerRegistryBootstrapServiceTest.java

```diff
@@ -0,0 +1,78 @@
+package com.netflix.eureka2.server.service.bootstrap;
+
+import com.netflix.eureka2.Server;
+import com.netflix.eureka2.interests.ChangeNotification;
+import com.netflix.eureka2.interests.ChangeNotification.Kind;
+import com.netflix.eureka2.interests.Interest;
+import com.netflix.eureka2.interests.Interests;
+import com.netflix.eureka2.interests.StreamStateNotification;
+import com.netflix.eureka2.metric.EurekaRegistryMetricFactory;
+import com.netflix.eureka2.registry.Source;
+import com.netflix.eureka2.registry.Source.Origin;
+import com.netflix.eureka2.registry.SourcedEurekaRegistry;
+import com.netflix.eureka2.registry.SourcedEurekaRegistryImpl;
+import com.netflix.eureka2.registry.instance.InstanceInfo;
+import com.netflix.eureka2.server.resolver.EurekaEndpoint;
+import com.netflix.eureka2.server.resolver.EurekaEndpointResolver;
+import com.netflix.eureka2.testkit.data.builder.SampleInstanceInfo;
+import org.junit.Before;
+import org.junit.Test;
+import rx.Observable;
+import rx.schedulers.Schedulers;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Tomasz Bak
+ */
+public class PeerRegistryBootstrapServiceTest {
```

Comment:
Would be nice to have additional tests to cover:
- bootstrap fail on first source but succeeds on subsequent source